docs/USAGE-GUIDE.md

```diff
@@ -720,45 +674,11 @@ TLS records until the provided output buffer is full.
 Unlike OpenSSL, repeated calls to `s2n_recv()` should not duplicate the original parameters,
 but should update the inputs per the indication of size read.
 
-For example, to read all data sent by the peer into one buffer:
-```c
-char buffer[MAX_BUFFER_SIZE] = { 0 };
-
-s2n_blocked_status blocked = S2N_NOT_BLOCKED;
-int bytes_read = 0;
-do {
-    int r = s2n_recv(conn, buffer + bytes_read, sizeof(buffer) - bytes_read, &blocked);
-    if (r < 0 && s2n_error_get_type(s2n_errno) != S2N_ERR_T_BLOCKED) {
-        printf("Error: %s. %s", s2n_strerror(s2n_errno, NULL), s2n_strerror_debug(s2n_errno, NULL));
-        break;
-    }
-    bytes_read += r;
-} while (blocked != S2N_NOT_BLOCKED);
-printf("Received: %.*s", bytes_read, buffer);
-```
-
-Careful: `bytes_read < sizeof(buffer)` is not a valid loop condition here.
-If the peer indicates end-of-data, `s2n_recv()` will begin returning 0 and you
-may enter an infinite loop.
+For an example of how to read all the data sent by the peer into one buffer,
+see `s2n_example_recv()` in [examples/s2n_recv.c](https://github.com/aws/s2n-tls/blob/main/docs/examples/s2n_recv.c)
```

Comment:
Yeah I agree that it's better to have the examples in separate folders so you can reference them separately.